cgrok/grok_discover.c

```diff
@@ -186,6 +186,8 @@ void grok_discover(const grok_discover_t *gdt, /*grok_t *dest_grok, */
     if (max_matchlen == 0) { /* No valid matches were found */
       if (first_match_endpos > 0) {
         offset += first_match_endpos;
+      } else {
```

Comment:
This resolves a super confidence-inspiring issue that was fixed in master but has yet to make it into any release: https://github.com/jordansissel/grok/commit/25acb06deae35072d530285f505a1a428aedf734